File: MB-Secretaria/production-safety/safe-fallback.ts
import type { RuntimeGuardResult } from "./types";

export function buildSafeFallback(reason: string[], guard?: Partial<RuntimeGuardResult>) {
  return {
    fallback_required: true,
    reason,
    action: "continue_existing_runtime_flow",
    blocks_critical_apis: false,
    dry_run: guard?.dry_run ?? true,
    shadow_mode: guard?.shadow_mode ?? true,
  };
}

File: api/app/runtime_support/production_safety/safe_fallback.py
from typing import Any


def build_safe_fallback(reason: list[str], guard: dict[str, Any] | None = None) -> dict[str, Any]:
    return {
        "fallback_required": True,
        "reason": reason,
        "action": "continue_existing_runtime_flow",
        "blocks_critical_apis": False,
        "dry_run": (guard or {}).get("dry_run", True),
        "shadow_mode": (guard or {}).get("shadow_mode", True),
    }
